core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,98 @@
+package com.baeldung.inputstream;
+
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamUnitTest {
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File sampleOutFile = sampleOut.toFile();
+        try (InputStream inputStream = new FileInputStream(sampleOutFile)) {
+            Assert.assertTrue(readString(inputStream).contains(lines.get(0)));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToFileInputStreamWithStringConstructor_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        String expectedText = "Hello. Hi.";
+        List<String> lines = Arrays.asList(expectedText);
+        Files.write(sampleOut, lines);
+        String fileAbsolutePath = sampleOut.toFile()
+            .getAbsolutePath();
+        try (FileInputStream fis = new FileInputStream(fileAbsolutePath)) {
+            int content;
+            int availabeBytes = fis.available();
+            Assert.assertTrue(availabeBytes > 0);
+            StringBuilder actualReadText = new StringBuilder();
+            while ((content = fis.read()) != -1) {
+                actualReadText.append((char) content);
+            }
+            Assert.assertTrue(actualReadText.toString()
+                .contains(expectedText));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToByteArrayInputStream_thenShouldMatchWhenRead() throws IOException {
```

Comment:
Test name does not match what it is doing. The test reads a string from a stream, there is no writing happening here?